### PR TITLE
Device status improvements

### PR DESCRIPTION
**Pull request by ps2**
New swift structures to represent Nightscout device status upload api json.  It's infrastructure for uploading Loop status to NS, and also for uploading phone status separately from pump status.

### CODE REVIEW

File: NightscoutUploadKit/DeviceStatus/BatteryStatus.swift [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+//  BatteryStatus.swift
+//  RileyLink
+//
+//  Created by Pete Schwamb on 7/28/16.
+//  Copyright © 2016 Pete Schwamb. All rights reserved.
+//
+
+import Foundation
+
+public class BatteryStatus {
```
Comment (loudnate) [inline NightscoutUploadKit/DeviceStatus/BatteryStatus.swift]: Any reason to use classes rather than structs?

**Comment by loudnate** [comment]
LGTM

File: NightscoutUploadKit/DeviceStatus/BatteryStatus.swift (hunk 1 above)
Comment (ps2) [inline NightscoutUploadKit/DeviceStatus/BatteryStatus.swift]: Just that I haven't been indoctrinated fully.  Getting closer, thanks to you. :)

File: NightscoutUploadKit/DeviceStatus/IOBStatus.swift [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+//
+//  Created by Pete Schwamb on 7/28/16.
+//  Copyright © 2016 Pete Schwamb. All rights reserved.
+//
+
+import Foundation
+
+public struct IOBStatus {
+    let iob: Double // basal iob + bolus iob: can be negative
+    let basaliob: Double
```
Comment (loudnate) [inline NightscoutUploadKit/DeviceStatus/IOBStatus.swift:13]: `basalIOB` would be more cocoa-like.

**Pull request merged**